core/state_processor.go

```diff
@@ -76,6 +77,24 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		vmenv   = vm.NewEVM(context, vm.TxContext{}, statedb, p.config, cfg)
 		signer  = types.MakeSigner(p.config, header.Number, header.Time)
 	)
+	if beaconRoot != nil {
+		// If EIP-4788 is enabled, we need to invoke the beaconroot storage contract with
+		// the new root
+		// TODO: Right now, I set the GasXX fields to 0. If this does not work,
+		// we can use the header.BaseFee and set that instead.
+		msg := &Message{
+			From:      params.SystemAddress,
+			GasLimit:  100_000,
```

Comment:
I suggest to use gas limit of 30_000_000 to be in line with the Gnosis Chain [spec](https://github.com/gnosischain/specs/blob/master/execution/withdrawals.md#specification) for system transactions.